tools/scoring/score/dimensions/dimension.py

```diff
@@ -176,8 +180,123 @@ def is_entity_virtual(*, entity: EntityInstance) -> bool:
     """
     return entity.links is not None
 
+  @staticmethod
+  def match_virtual_entities(
+      *, solution_points_virtual: List[Tuple[Set[str], EntityType]],
+      proposed_points_virtual: List[Tuple[Set[str], EntityType]],
+      sort_candidates_by_key: str) -> Dict[float, List[Dict]]:
+    """
+    Finds the closest correlating virtual entities between two files.
+
+    Args:
+      solution_points_virtual: Raw field names and entity types
+        for all virtual entities in a file
+      proposed_points_virtual: Raw field names and entity types
+        for all virtual entities in a file
+      sort_candidates_by_key: Parameter by which to "break a tie"
+        if there are multiple matches with the same subscore.
+
+    Returns:
+      Dictionary whose keys are floats representing the extent to which the
```

Comment:
I am not sure I understand this part.
create a structure?

Reply:
Good idea.